src/gc/impl/conservative/gc.d

```diff
@@ -2563,11 +2580,17 @@ struct Gcx
         return recoverPool[bin] = poolIndex < npools ? cast(SmallObjectPool*)pool : null;
     }
 
+    void disableFork() nothrow
+    {
+        config.fork = false;
+        markProcPid = 0;
```

Comment:
Thinking about the case where fork can be enabled and disabled freely at runtime, here it should be checked if a concurrent mark is running. I do not recommend allowing that though, not at least without a deep review of the code as the original implementation didn't consider this option.

Reply:
If config is not modified by the GC but keeps its own flag, `config.fork` can just be checked before starting the next collection. That won't cause any racing conditions.